utils/fixed_decimal/src/decimal.rs

```
@@ -697,6 +697,395 @@ impl FromStr for FixedDecimal {
+    /// Specify that the floating point number is precise to the maximum representable by IEEE.
+    ///
+    /// This results in a FixedDecimal having enough digits to recover the original floating point
+    /// value, with no trailing zeros.
+    Maximum,
+}
+
+#[cfg(feature = "ryu")]
+impl FixedDecimal {
+    /// Construct a [`FixedDecimal`] from an f64. This uses `ryu` and
+    /// goes through an intermediate string representation, so is not
+    /// fully efficient. See https://github.com/unicode-org/icu4x/issues/166 for
+    /// more details.
+    ///
+    /// This function can be made available with the `"ryu"` feature.
+    ///
+    /// ```rust
+    /// use fixed_decimal::{DoublePrecision, FixedDecimal};
+    /// use writeable::Writeable;
+    ///
+    /// let decimal = FixedDecimal::new_from_f64(0.012345678, DoublePrecision::Maximum).unwrap();
+    /// assert_eq!(decimal.writeable_to_string(), "0.012345678");
+    ///
+    /// let decimal = FixedDecimal::new_from_f64(-123456.78, DoublePrecision::Maximum).unwrap();
+    /// assert_eq!(decimal.writeable_to_string(), "-123456.78");
+    ///
+    /// let decimal = FixedDecimal::new_from_f64(12345678000., DoublePrecision::Maximum).unwrap();
+    /// assert_eq!(decimal.writeable_to_string(), "12345678000.0");
+    /// ```
+    pub fn new_from_f64(float: f64, precision: DoublePrecision) -> Result<Self, Error> {
+        let mut decimal = Self::new_from_f64_raw(float)?;
+        match precision {
+            DoublePrecision::Maximum => (),
+            DoublePrecision::Magnitude(mag) => {
+                let n_digits = decimal.digits.len() as i16;
+                // magnitude of the lowest digit in self.digits
+                let lowest_magnitude = decimal.magnitude - n_digits + 1;
+
+                if mag > lowest_magnitude {
+                    let round_by = (mag - lowest_magnitude) as i16;
+
+                    if round_by <= n_digits {
+                        decimal.round_digits(round_by as u16);
```

Suggestion: to avoid the `round_digits` function, you could return an error here for the time being (corresponding to `RoundingMode::Unnecessary`)